### PR TITLE
Enable nitpicky Sphinx warnings, fix issues and fail on new issues

### PR DESCRIPTION
**Pull request by Gallaecio**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py27, pypy, py34, py35, py36, py37, pypy3
+envlist = py27, pypy, py34, py35, py36, py37, pypy3, docs
```
Comment (kmike) [inline tox.ini]: Could you please enable it on Travis as well?

**Comment by Gallaecio** [comment]
Wow, the error seems too specific to be a false positive :thinking:

**Comment by kmike** [comment]
Yeah, looks like a genuine failure

**Comment by Gallaecio** [comment]
It’s a regression of https://bitbucket.org/pypy/pypy/issues/2389/different-behavior-of-bytesdecode-utf8 which @lopuhin reported to PyPy in 2016. His very example then can be used to reproduce the issue.

**Comment by Gallaecio** [comment]
@kmike If you are OK with running the tests on Python 2 PyPy 7.0.0 for now, we can squash and merge this.

**Comment by kmike** [comment]
Sounds good, thanks @Gallaecio!

**Pull request merged**